### PR TITLE
Port Randomly Fire Dropped Weapons

### PR DESCRIPTION
**Pull request by VMSolidus**
# Description

Ports https://github.com/Simple-Station/Parkstation-Friendly-Chainsaw/pull/19

I have added a new stat for firearms, the "Reliability" stat, which is a number between 0 and 1. It's used as a percentage chance for the weapon to fire itself when violently thrown into anyone. This PR differs from the original one slightly in that to get it to actually work without crashing, I set the system to listen to an event that triggers whenever the gun collides with another entity, not necessarily just the floor. This is the same event responsible for the clown's cream pie system, or for glass shards embedding in an entity.

# Changelog

:cl:
- add: NanoTrasen has disabled the unneeded safeties on your guns- Make sure you're careful with them!
- tweak: All Firearms now have a reliability stat, some are more reliable than others. The more reliable a weapon is, the less likely it is to accidentally discharge when yeeted.

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Pistols/pistols.yml [hunk 1]
```diff
@@ -198,7 +209,7 @@
   name: N1984
   parent: BaseWeaponPistol
   id: WeaponPistolN1984 # the spaces in description are for formatting.
-  description: The sidearm of any self respecting officer.     Comes in .45 magnum, the lord's caliber. 
+  description: The sidearm of any self respecting officer.     Comes in .45 magnum, the lord's caliber.
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Pistols/pistols.yml:212]: :O

File: Content.Shared/Weapons/Ranged/Components/GunComponent.cs [hunk 2]
```diff
@@ -229,6 +229,12 @@ public sealed partial class GunComponent : Component
     /// </summary>
     [DataField]
     public bool ClumsyProof = false;
+
+    /// <summary>
+    ///     The percentage chance of a given gun to accidentally discharge if violently thrown into a wall or person
+    /// </summary>
+    [DataField]
+    public float Reliability = 0.1f;
```
Comment (DangerRevolution) [inline Content.Shared/Weapons/Ranged/Components/GunComponent.cs:237]: So higher reliability the worse?

**Comment by DangerRevolution** [comment]
Just going to play with this in game pre-approvd

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Pistols/pistols.yml (hunk 1 above)
Comment (VMSolidus) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Pistols/pistols.yml:212]: Lol, end of line removals. Ignore it, this is supposed to happen.

File: Content.Shared/Weapons/Ranged/Components/GunComponent.cs (hunk 2 above)
Comment (VMSolidus) [reply Content.Shared/Weapons/Ranged/Components/GunComponent.cs:237]: Correct. :trollface:

Comment (DEATHB4DEFEAT) [reply Content.Shared/Weapons/Ranged/Components/GunComponent.cs:237]: Should have a more clear name.

**Review by DEATHB4DEFEAT — APPROVED** [review]

File: Content.Shared/Weapons/Ranged/Components/GunComponent.cs (hunk 2 above)
Comment (VMSolidus) [reply Content.Shared/Weapons/Ranged/Components/GunComponent.cs]: I changed the name to FireOnDropChance, so that its much clearer what it does.

**Review by DangerRevolution — APPROVED** [review]
some of the stats are a little sussy but we can kick it down the road for an eventual .yaml cleanup and when common core becomes a thing

**Pull request merged**